range.c

```diff
@@ -1017,6 +1017,31 @@ range_to_a(VALUE range)
     return rb_call_super(0, 0);
 }
 
+/*
+ *  call-seq:
+ *    to_set -> set
+ *
+ *  Returns a set containing the elements in +self+, if a finite collection;
+ *  raises an exception otherwise.
+ *
+ *    (1..4).to_set     # => Set[1, 2, 3, 4]
+ *    (1...4).to_set    # => Set[1, 2, 3]
+ *    ('a'..'d').to_set # => Set["a", "b", "c", "d"]
+ *    (..'d').to_set    # RangeError
+ *    ('a'..).to_set    # RangeError
+ *
+ */
+
+
+static VALUE
+range_to_set(VALUE range)
```

Comment:
Although passing arguments to `Enumerable#to_set` is deprecated, it is allowed yet.
Shouldn't this consistent with it too?